src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -401,6 +405,28 @@ public static function observe($class, $priority = 0)
     }
 
     /**
+     * Merge the given attributes with those from the request.
+     *
+     * @param array                         $attributes
+     * @param \Illuminate\Http\Request|null $request
+     * @return array
+     */
+    protected function merge(array $attributes = [], Request $request = null)
+    {
+        if (is_null($request)) {
+            return $attributes;
+        }
+
+        $fromRequest = $request->input(class_basename($this));
```

Comment:
In case anyone's wondering why I choose to search for the class basename: it's because that still works if you have a form that saves two different models which happen to have an attirbute with the same attribute name. Like `child.name` and `school.name` for example.